### PR TITLE
doc: Added changelog

### LINKED ISSUE #240: Create Changelog
Creating a possibility to save/archive a changes of features and APIs of Mantella.

### PR DESCRIPTION
**Pull request by dkiechle**
fixed #240 

My contribution is licensed under the MIT license.

### CODE REVIEW

**Comment by SebastianNiemann** [comment]
Please resolve the conflicts introduced by #311 first.

**Comment by dkiechle** [comment]
Is already finished waiting of #317

**Comment by SebastianNiemann** [comment]
@dkiechle I introduced new conflicts with #317 (sorry :sweat_smile:)

**Comment by dkiechle** [comment]
@SebastianNiemann You can merge this pull request not before fix #318

**Comment by SebastianNiemann** [comment]
@dkiechle Please solve the conflicts

File: doc/.pre_processing.py [hunk 1]
```diff
@@ -192,6 +213,37 @@ class Colors:
     docfile.close()
     # Clears the last written output (the whole line)
     print('\x1b[2K \r', end="")
+    
+# Generate changelog
+## sort reverse for actual version on top
```
Comment (SebastianNiemann) [inline doc/.pre_processing.py]: Traverses the changes in reverse order, to list the latest changes on top

File: doc/.pre_processing.py [hunk 2]
```diff
@@ -192,6 +213,37 @@ class Colors:
     docfile.close()
     # Clears the last written output (the whole line)
     print('\x1b[2K \r', end="")
+    
+# Generate changelog
```
Comment (SebastianNiemann) [inline doc/.pre_processing.py]: Generates the changelog

File: doc/.pre_processing.py [hunk 3]
```diff
@@ -71,6 +91,7 @@ class Colors:
   # Create subdirectory if missing
   os.makedirs(os.path.dirname(file[1]), exist_ok=True)
```
Comment (SebastianNiemann) [inline doc/.pre_processing.py]: Having the `*.include` files, is this even possible anymore?

**Comment by SebastianNiemann** [comment]
There are also a few cases (in old comment blocks), that need to be improved. I added an additional issue for these (#350).

**Pull request merged**